### PR TITLE
Add a Riemann reporter - Work in Progress

### PR DESCRIPTION
**Pull request by Ben-M**
There's still work to be done here (I'll describe it in the comments), but I'm sending the PR now to get your input early.

### CODE REVIEW

**Comment by Ben-M** [comment]
We need to decide how to deal with dependencies - do we always want to require the Riemann gem, or do we want to tell people to require 'vitals/riemann' and add the Riemann client to their Gemfile?

**Comment by Ben-M** [comment]
I need to update the Readme

File: lib/vitals/reporters/riemann_reporter.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require 'riemann/client'
+
+module Vitals::Reporters
+  class RiemannReporter < BaseReporter
+    attr_accessor :format
+    attr_reader :riemann
+
+    def initialize(host: 'localhost', port: 5555, timeout: 5, format:nil)
+      @riemann = Riemann::Client.new(host: host, port: port, timeout: timeout)
```
Comment (Ben-M) [inline lib/vitals/reporters/riemann_reporter.rb]: The Riemann client does not open the connection(which is slowish) until the first metric is sent. Do we want to warm the connection by sending a 'connected' even to Riemann?

**Comment by jondot** [comment]
Thanks
If you'll notice, you didn't touch vitals core. That's due to vitals' design for extensibility. So actually you're already looking at how plugins are made -
just make a vitals-riemann gem instead and take vitals as a dep.
LMK if you new any help with that

**Comment by Ben-M** [comment]
Done. 

https://github.com/Ben-M/vitals-riemann_reporter

**Pull request closed without merging**